### PR TITLE
treewide: fetchFromGitHub

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Description of changes
switch from `fetchgit` to `fetchFromGitHub`

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Stunkymonkey** [comment]
Result of `nixpkgs-review pr 163582` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>50 packages built:</summary>
  <ul>
    <li>Literate</li>
    <li>adb-sync</li>
    <li>alock</li>
    <li>azure-vhd-utils</li>
    <li>bash-my-aws</li>
    <li>brickd</li>
    <li>bud</li>
    <li>codesearch</li>
    <li>das_watchdog</li>
    <li>dconf2nix</li>
    <li>diffoscope</li>
    <li>diffoscopeMinimal</li>
    <li>gawp</li>
    <li>go-mtpfs</li>
    <li>go-repo-root</li>
    <li>go-upower-notify</li>
    <li>gosu</li>
    <li>gotags</li>
    <li>gvolicon</li>
    <li>i3cat</li>
    <li>jd</li>
    <li>lightdm-enso-os-greeter</li>
    <li>lightum</li>
    <li>morty</li>
    <li>ps3netsrv</li>
    <li>python310Packages.editorconfig</li>
    <li>python310Packages.jsbeautifier</li>
    <li>python310Packages.mutag</li>
    <li>python310Packages.samplerate</li>
    <li>python39Packages.editorconfig</li>
    <li>python39Packages.jsbeautifier</li>
    <li>python39Packages.mutag</li>
    <li>python39Packages.samplerate</li>
    <li>rt5677-firmware</li>
    <li>rtags</li>
    <li>s3gof3r</li>
    <li>sdl-jstest</li>
    <li>svg2tikz</li>
    <li>tpm-luks</li>
    <li>trustedGrub</li>
    <li>trustedGrub-for-HP</li>
    <li>ua</li>
    <li>usermount</li>
    <li>vimPlugins.YouCompleteMe</li>
    <li>vimPlugins.vim-go</li>
    <li>vobsub2srt</li>
    <li>waitron</li>
    <li>ws</li>
    <li>xtrlock-pam</li>
    <li>ycmd</li>
  </ul>
</details>

File: pkgs/tools/text/codesearch/default.nix [hunk 1]
```diff
@@ -1,24 +1,23 @@
...
-  meta = {
+  meta = with lib; {
     description = "Fast, indexed regexp search over large file trees";
     homepage = "https://github.com/google/codesearch";
-    license = [ lib.licenses.bsd3 ];
-    maintainers = [ lib.maintainers.bennofs ];
-    platforms = lib.platforms.unix;
+    license = [ licenses.bsd3 ];
+    maintainers = with maintainers; [ bennofs ];
+    platforms = platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/tools/text/codesearch/default.nix:21]:
```suggestion
```

File: pkgs/tools/networking/ua/default.nix [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
 
-  meta = {
+  meta = with lib; {
     homepage = "https://github.com/sloonz/ua";
-    license = lib.licenses.isc;
+    license = licenses.isc;
     description = "Universal Aggregator";
-    platforms = lib.platforms.unix;
-    maintainers = with lib.maintainers; [ ttuegel ];
+    platforms = platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/ua/default.nix:28]:
```suggestion
```

File: pkgs/tools/networking/s3gof3r/default.nix [hunk 3]
```diff
@@ -1,17 +1,21 @@
...
+    owner = "rlmcpherson";
+    repo = "s3gof3r";
+    rev = "31603a0dc94aefb822bfe2ceea75a6be6013b445";
+    sha256 = "sha256-88C6c4DRD/4ePTO1+1YiI8ApXWE2uUlr07dDCxGzaoE=";
   };
 
   goDeps = ./deps.nix;
+
+  meta = with lib; {
+    homepage = "https://pkg.go.dev/github.com/rlmcpherson/s3gof3r";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/s3gof3r/default.nix:19]: We should also set the other meta things.

File: pkgs/tools/misc/upower-notify/default.nix [hunk 4]
```diff
@@ -8,16 +8,20 @@
...
+    owner = "omeid";
+    repo = "upower-notify";
+    rev = "14c581e683a7e90ec9fa6d409413c16599a5323c";
     sha256 = "16zlvn53p9m10ph8n9gps51fkkvl6sf4afdzni6azk05j0ng49jw";
   };
 
   goDeps = ./deps.nix;
+
+  meta = with lib; {
+    homepage = "https://github.com/omeid/upower-notify";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/upower-notify/default.nix:25]: We should also set the other meta things.

File: pkgs/tools/misc/upower-notify/default.nix [hunk 5]
```diff
@@ -1,4 +1,4 @@
-{ lib, buildGoPackage, fetchgit }:
+{ lib, buildGoPackage, fetchFromGitHub }:
 
 # To use upower-notify, the maintainer suggests adding something like this to your configuration.nix:
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/upower-notify/default.nix:3]: Which maintainer?...

File: pkgs/tools/misc/i3cat/default.nix [hunk 6]
```diff
@@ -1,17 +1,21 @@
...
+    owner = "vincent-petithory";
+    repo = "i3cat";
+    rev = "v${version}";
+    sha256 = "sha256-BxiiYzSjvXAMUQSUTKviLvrmGjkCLW6QPrgBBHvvF+Q=";
   };
 
   goDeps = ./deps.nix;
+
+  meta = with lib; {
+    homepage = "https://vincent-petithory.github.io/i3cat/";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/i3cat/default.nix:19]: We should also set the other meta things.

File: pkgs/tools/misc/gawp/default.nix [hunk 7]
```diff
@@ -1,18 +1,18 @@
-{ lib, buildGoPackage, fetchgit }:
+{ lib, buildGoPackage, fetchFromGitHub }:
 
 with builtins;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/gawp/default.nix:4]:
```suggestion
```
Unused

File: pkgs/servers/brickd/default.nix [hunk 8]
```diff
@@ -1,24 +1,24 @@
...
   inherit version;
 
-  src = fetchgit {
-    url = "git://github.com/Tinkerforge/brickd.git";
-    rev = "refs/tags/v${version}";
-    sha256 = "0m2q01sbgf8z4559jpr6k3jivb8x98vxv1fhgx8nfcjbwz1q83gb";
+  src = fetchFromGitHub {
+    owner = "Tinkerforge";
+    repo = "brickd";
+    rev = "v2.1.1";
```
Comment (SuperSandro2000) [inline pkgs/servers/brickd/default.nix:20]:
```suggestion
    rev = "v${version}";
```

File: pkgs/servers/brickd/default.nix [hunk 9]
```diff
@@ -1,24 +1,24 @@
...
-version = "2.1.1";
-daemonlib = fetchgit {
-    url = "https://github.com/Tinkerforge/daemonlib.git";
-    rev = "refs/tags/brickd-${version}";
-    sha256 = "097kaz7d0rzg0ijvcna3y620k3m5fgxpqsac5gbhah8pd7vlj1a4";
+  version = "2.1.1";
+  daemonlib = fetchFromGitHub {
+    owner = "Tinkerforge";
+    repo = "daemonlib";
+    rev = "brickd-2.1.1";
```
Comment (SuperSandro2000) [inline pkgs/servers/brickd/default.nix:8]:
```suggestion
    rev = "brickd-${version}";
```

File: pkgs/development/tools/rtags/default.nix [hunk 10]
```diff
@@ -8,18 +8,20 @@ stdenv.mkDerivation rec {
...
-    url = "https://github.com/andersbakken/rtags.git";
-    sha256 = "1iwvp7a69sj3wqjgcnyh581qrpicxzi2lfjkxqpabpyjkl5nk7hh";
     # unicode file names lead to different checksums on HFS+ vs. other
     # filesystems because of unicode normalisation
     postFetch = ''
       rm $out/src/rct/tests/testfile_*.txt
     '';
   };
 
+
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rtags/default.nix:24]:
```suggestion
```

File: pkgs/development/python-modules/editorconfig/default.nix [hunk 11]
```diff
@@ -1,20 +1,22 @@
...
+  src = fetchFromGitHub {
+    owner = "editorconfig";
+    repo = "editorconfig-core-py";
     rev = "1a8fb62b9941fded9e4fb83a3d0599427f5484cb"; # Not tagged
-    sha256 = "0vx8rl7kii72965jsi01mdsz9rfi1q9bwy13x47iaqm6rmcwc1rb";
+    sha256 = "sha256-KwfGWc2mYhUP6SN4vhIO0eX0dasBRC2LSeLEOA/NqG8=";
+    fetchSubmodules = true; # to ensure test submodule is available
   };
 
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/editorconfig/default.nix:19]:
```suggestion
```

**Comment by Stunkymonkey** [comment]
Result of `nixpkgs-review pr 163582` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>50 packages built:</summary>
  <ul>
    <li>Literate</li>
    <li>adb-sync</li>
    <li>alock</li>
    <li>azure-vhd-utils</li>
    <li>bash-my-aws</li>
    <li>brickd</li>
    <li>bud</li>
    <li>codesearch</li>
    <li>das_watchdog</li>
    <li>dconf2nix</li>
    <li>diffoscope</li>
    <li>diffoscopeMinimal</li>
    <li>gawp</li>
    <li>go-mtpfs</li>
    <li>go-repo-root</li>
    <li>go-upower-notify</li>
    <li>gosu</li>
    <li>gotags</li>
    <li>gvolicon</li>
    <li>i3cat</li>
    <li>jd</li>
    <li>lightdm-enso-os-greeter</li>
    <li>lightum</li>
    <li>morty</li>
    <li>ps3netsrv</li>
    <li>python310Packages.editorconfig</li>
    <li>python310Packages.jsbeautifier</li>
    <li>python310Packages.mutag</li>
    <li>python310Packages.samplerate</li>
    <li>python39Packages.editorconfig</li>
    <li>python39Packages.jsbeautifier</li>
    <li>python39Packages.mutag</li>
    <li>python39Packages.samplerate</li>
    <li>rt5677-firmware</li>
    <li>rtags</li>
    <li>s3gof3r</li>
    <li>sdl-jstest</li>
    <li>svg2tikz</li>
    <li>tpm-luks</li>
    <li>trustedGrub</li>
    <li>trustedGrub-for-HP</li>
    <li>ua</li>
    <li>usermount</li>
    <li>vimPlugins.YouCompleteMe</li>
    <li>vimPlugins.vim-go</li>
    <li>vobsub2srt</li>
    <li>waitron</li>
    <li>ws</li>
    <li>xtrlock-pam</li>
    <li>ycmd</li>
  </ul>
</details>

File: pkgs/development/python-modules/editorconfig/default.nix [hunk 12]
```diff
@@ -1,18 +1,19 @@
...
   pname = "editorconfig";
   version = "0.12.3";
 
-  # fetchgit used to ensure test submodule is available
-  src = fetchgit {
-    url = "https://github.com/editorconfig/editorconfig-core-py";
+  src = fetchFromGitHub {
+    owner = "editorconfig";
+    repo = "editorconfig-core-py";
     rev = "1a8fb62b9941fded9e4fb83a3d0599427f5484cb"; # Not tagged
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/editorconfig/default.nix:14]:
```suggestion
    rev = "v${version}";
```

File: pkgs/tools/networking/s3gof3r/default.nix [hunk 13]
```diff
@@ -1,17 +1,24 @@
-{ lib, buildGoPackage, fetchgit }:
+{ lib, buildGoPackage, fetchFromGitHub }:
 
 buildGoPackage rec {
   pname = "s3gof3r";
-  version = "20151109-${lib.strings.substring 0 7 rev}";
-  rev = "31603a0dc94aefb822bfe2ceea75a6be6013b445";
+  version = "unstable-2015-11-09";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/s3gof3r/default.nix:5]:
```suggestion
  version = "0.5.0";
```

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 163582` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>Literate</li>
    <li>diffoscope (diffoscopeMinimal)</li>
    <li>python310Packages.editorconfig</li>
    <li>python310Packages.jsbeautifier</li>
    <li>python39Packages.editorconfig</li>
    <li>python39Packages.jsbeautifier</li>
  </ul>
</details>
<details>
  <summary>26 packages built:</summary>
  <ul>
    <li>azure-vhd-utils</li>
    <li>bash-my-aws</li>
    <li>codesearch</li>
    <li>dconf2nix</li>
    <li>gawp</li>
    <li>go-mtpfs</li>
    <li>go-repo-root</li>
    <li>go-upower-notify</li>
    <li>gotags</li>
    <li>i3cat</li>
    <li>jd</li>
    <li>morty</li>
    <li>ps3netsrv</li>
    <li>python310Packages.mutag</li>
    <li>python310Packages.samplerate</li>
    <li>python39Packages.mutag</li>
    <li>python39Packages.samplerate</li>
    <li>rtags</li>
    <li>s3gof3r</li>
    <li>svg2tikz</li>
    <li>ua</li>
    <li>vimPlugins.YouCompleteMe</li>
    <li>vimPlugins.vim-go</li>
    <li>vobsub2srt</li>
    <li>ws</li>
    <li>ycmd</li>
  </ul>
</details>

**Pull request merged**